src/test/java/jenkins/plugins/http_request/HttpRequestRoundTripTest.java

```diff
@@ -0,0 +1,174 @@
+          assertEquals(bnvp.getValue(),anvp.getValue());
+        }
+
+        // Basic authentication check
+        List<BasicDigestAuthentication> beforeBdas = before.getDescriptor().getBasicDigestAuthentications();
+        List<BasicDigestAuthentication> afterBdas  = after.getDescriptor().getBasicDigestAuthentications();
+        assertEquals(beforeBdas.size(), afterBdas.size());
+        for (int idx = 0; idx < beforeBdas.size(); idx++) {
+            BasicDigestAuthentication beforeBda = beforeBdas.get(idx);
+            BasicDigestAuthentication afterBda = afterBdas.get(idx);
+            assertEquals(beforeBda.getKeyName(), afterBda.getKeyName());
+            assertEquals(beforeBda.getUserName(),afterBda.getUserName());
+            assertEquals(beforeBda.getPassword(),afterBda.getPassword());
+        }
+
+        // Form authentication check
+        List<FormAuthentication> beforeFas = before.getDescriptor().getFormAuthentications();
+        List<FormAuthentication> afterFas  = after.getDescriptor().getFormAuthentications();
+        assertEquals(beforeFas.size(), afterFas.size());
+        for (int idx = 0; idx < beforeFas.size(); idx++) {
+            FormAuthentication beforeFa = beforeFas.get(idx);
+            FormAuthentication afterFa  = afterFas.get(idx);
+            assertEquals(beforeFa.getKeyName(), afterFa.getKeyName());
+            List<RequestAction> beforeActions = beforeFa.getActions();
+            List<RequestAction> afterActions  = afterFa.getActions();
+            assertEquals(beforeActions.size(), afterActions.size());
+            for (int jdx = 0; jdx < beforeActions.size(); jdx ++) {
+                RequestAction beforeAction = beforeActions.get(jdx);
+                RequestAction afterAction  = afterActions.get(jdx);
+                assertEquals(beforeAction.getUrl(), afterAction.getUrl());
+                assertEquals(beforeAction.getMode(), afterAction.getMode());
+                List<NameValuePair> beforeParams = beforeAction.getParams();
+                List<NameValuePair> afterParams  = afterAction.getParams();
+                assertEquals(beforeParams.size(), afterParams.size());
+                for (int kdx = 0; kdx < beforeParams.size(); kdx++) {
+                    NameValuePair beforeNvp = beforeParams.get(kdx);
+                    NameValuePair afterNvp  = afterParams.get(kdx);
+                    assertEquals(beforeNvp.getName(), afterNvp.getName());
+                    assertEquals(beforeNvp.getValue(), afterNvp.getValue());
+                }
+            }
+        }
+    }
+
+    private <T extends Builder> T doRoundTrip(T before, Class<T> clazz) throws Exception {
+        FreeStyleProject p = j.createFreeStyleProject();
+        p.getBuildersList().add(before);
+
+        j.submit(j.createWebClient().getPage(p,"configure").getFormByName("config"));
+        T after = p.getBuildersList().get(clazz);
+        return after;
```

Comment:
Inspired by: https://github.com/jenkinsci/htmlpublisher-plugin/blob/master/src/test/java/htmlpublisher/workflow/PublishHTMLStepTest.java#L204

I'll try to find another example. If you know of one, let me know.